app/controllers/incomeSources/add/AddIncomeSourceStartDateController.scala

```diff
@@ -81,72 +81,86 @@ class AddIncomeSourceStartDateController @Inject()(authenticate: AuthenticationP
     handleSubmitRequest(
       incomeSourceType = incomeSourceType,
       isAgent = isAgent,
-      isUpdate = isChange
+      isChange = isChange
     )
   }
 
   private def handleShowRequest(incomeSourceType: IncomeSourceType,
                                 isAgent: Boolean,
-                                isUpdate: Boolean)
+                                isChange: Boolean)
                                (implicit user: MtdItUser[_]): Future[Result] = {
 
     val messagesPrefix = incomeSourceType.startDateMessagesPrefix
 
-      if (isEnabled(IncomeSources)) {
-        getFilledForm(form(messagesPrefix), incomeSourceType, isUpdate) map {
-          case Right(form) =>
-            Ok(
-              addIncomeSourceStartDate(
-                form = form,
-                isAgent = isAgent,
-                messagesPrefix = messagesPrefix,
-                backUrl = getBackUrl(incomeSourceType, isAgent, isUpdate),
-                postAction = getPostAction(incomeSourceType, isAgent, isUpdate)
-              )
-            )
-          case Left(ex) =>
-            Logger("application").error(s"[AddIncomeSourceStartDateController][handleShowRequest]: " +
-              s"Failed to get income source start date from session, reason: ${ex.getMessage}")
-            showInternalServerError(isAgent)
+    withIncomeSourcesFS {
+      if (!isChange && incomeSourceType.equals(UkProperty) || !isChange && incomeSourceType.equals(ForeignProperty)) {
```

Comment:
could refactor this to a match on a group eg (!isChange, isUkProp || isForProp) match {..